### PR TITLE
"lazy load" CSS

### PR DESCRIPTION
**Pull request by WMDE-Fisch**
moved CSS for hint-boxes and throbbers to separate files
corresponding CSS will only be loaded if really needed

see https://phabricator.wikimedia.org/T144168

### CODE REVIEW

File: DeepCat.js [hunk 1]
```diff
@@ -562,14 +565,32 @@
...
-		var smallHintBox = '<div id="deepcat-smallhint">'
+		var smallHintBox = '<div id="deepcat-smallhint" style="display: none;">'
 			+ '<img id="deepcat-smallhint-hide" title="' + mw.msg( 'deepcat-smallhint-close' ) + '" src="https://upload.wikimedia.org/wikipedia/commons/4/44/Curation_bar_icon_close.png">'
 			+ mw.msg( 'deepcat-hintbox-small' )
 			+ '</div>';
 		$( '#searchform' ).after( smallHintBox );
 		$( '#deepcat-smallhint-hide' ).on( 'click', hideSmallHint );
 	}
 
+	function loadHintCodeIfNeeded() {
```
Comment (thiemowmde) [inline DeepCat.js]: This is not a good function name. It seems it does more than "loading code". See the `add…` calls below.

File: DeepCat.hintbox.css [hunk 2]
```diff
@@ -1,19 +1,8 @@
 /**
  * DeepCat Gadget for MediaWiki
  * @licence GNU GPL v2+
- * @author Christoph Fischer < christoph.fischer@wikimedia.de >
+ * @author Christoph Jauera < christoph.jauera@wikimedia.de >
```
Comment (thiemowmde) [inline DeepCat.hintbox.css]: This name change should be a separate patch, but this is not critical.

:+1: overall, looks good.

File: DeepCat.js (hunk 1 above)
Comment (WMDE-Fisch) [reply DeepCat.js]: Done.

File: DeepCat.hintbox.css (hunk 2 above)
Comment (WMDE-Fisch) [reply DeepCat.hintbox.css]: True. Will do it in an other patch :-)

**Comment by WMDE-Fisch** [comment]
/me pokes @manicki or @jakob-WMDE to have a 2nd look on this PR

File: DeepCat.js [hunk 3]
```diff
@@ -562,14 +565,32 @@
...
+		if( hintCodeLoaded ) {
+			return;
+		}
+		mw.loader.load( cssPath + 'DeepCat.hintbox.css&action=raw&ctype=text/css', 'text/css' );
+		addSearchFormHint();
+		addSmallFormHint();
+		hintCodeLoaded = true;
+	}
+
+	function loadThrobberCodeIfNeeded() {
```
Comment (manicki) [inline DeepCat.js]: `code` seems a bit mysterious, then I check what it does, and it load styles. Maybe `loadThrobberStyles` or `loadThrobberResources`? I am not sure if I'd put the "IfNeeded" part to the method names but it might be just matter of taste, I am not saying you should change it one way or another.

File: DeepCat.js [hunk 4]
```diff
@@ -562,14 +565,32 @@
...
-		var smallHintBox = '<div id="deepcat-smallhint">'
+		var smallHintBox = '<div id="deepcat-smallhint" style="display: none;">'
 			+ '<img id="deepcat-smallhint-hide" title="' + mw.msg( 'deepcat-smallhint-close' ) + '" src="https://upload.wikimedia.org/wikipedia/commons/4/44/Curation_bar_icon_close.png">'
 			+ mw.msg( 'deepcat-hintbox-small' )
 			+ '</div>';
 		$( '#searchform' ).after( smallHintBox );
 		$( '#deepcat-smallhint-hide' ).on( 'click', hideSmallHint );
 	}
 
+	function initializeHintCodeIfNeeded() {
```
Comment (manicki) [inline DeepCat.js]: On first glimpse at the code I am really confused what "hint code" is.

Looking at the method what it does, it loads css styles and adds some HTML elements. That's OK but which of those two is the actual "code"? Or is the "hint code" something different hidden somewhere deeper?
Wouldn't `initializeHintBoxes(IfNeeded)?` sound more clear in this case?

File: DeepCat.js [hunk 5]
```diff
@@ -562,14 +565,32 @@
...
+		if( hintCodeLoaded ) {
+			return;
+		}
+		mw.loader.load( cssPath + 'DeepCat.hintbox.css&action=raw&ctype=text/css', 'text/css' );
```
Comment (manicki) [inline DeepCat.js]: Don't we want to put those CSS paths in some config-like variable (on top of the file or so), so it is easier to customize and paths are not hidden somewhere deep in private methods?

**Review by manicki — COMMENTED** [review]
Had a very quick look. Mostly old man nagging about function names. Please consider if my comment on CSS paths is relevant here, though.

I'll have another look on the patch today and will then verify it is actually working. Generally looking fine for me, we might be able to have this merged today!

File: DeepCat.js (hunk 5 above)
Comment (WMDE-Fisch) [reply DeepCat.js]: atm the path is in a config-like var at the top of the file :-)

File: DeepCat.js (hunk 3 above)
Comment (WMDE-Fisch) [reply DeepCat.js]: good point will change that

File: DeepCat.js (hunk 4 above)
Comment (WMDE-Fisch) [reply DeepCat.js]: good point will change that as well :-)

File: DeepCat.js (hunk 5 above)
Comment (manicki) [reply DeepCat.js]: Bah, yeah. What I probably meant was having full paths (i.e. including filename) as config variables. But this indeed might not be necessary. Unless we believe it is likely to want to have those CSS files named different when gadget is copied over to some other place or whatever other evil things happens. But shortly: right, it is fine as it is now!

**Comment by manicki** [comment]
Looks good to me now!
Just spotted no newline at the end of the new throbber CSS file but I am not going to block this PR on such thing.
